1.3.2 build fails

Please briefly describe your problem and what output you expect. If you have a question, please don't use this form. Instead, ask on <https://stackoverflow.com/> or <https://community.rstudio.com/>.

Please include a minimal reproducible example (AKA a reprex). If you've never heard of a [reprex](https://reprex.tidyverse.org/) before, start by reading <https://www.tidyverse.org/help/#reprex>.

---

Session log attached.

```r
# insert reprex here
```
[RMariaDB.txt](https://github.com/user-attachments/files/15569053/RMariaDB.txt)

v1.3.1.9000 is the latest tag that builds without error. Here’s the log from the earliest one that fails.

Thanks,
Rob


> remotes::install_github(
+     repo    = "r-dbi/RMariaDB",
+     ref     = "v1.3.1.9001",
+     lib     = "~/R",
+     upgrade = "never",
+     force   = TRUE
+ )
+ Downloading GitHub repo ***@***.***
-- R CMD build -------------------------------------------------------------------------------------------------------------------------------------
v  checking for file '/tmp/RtmpIbXsV6/remotes72e02dff314f/r-dbi-RMariaDB-420fa50/DESCRIPTION'
-  preparing 'RMariaDB':
v  checking DESCRIPTION meta-information
-  cleaning src
-  running 'cleanup'
-  checking for LF line-endings in source and make files and shell scripts
-  checking for empty or unneeded directories
   Removed empty directory 'RMariaDB/.aviator'
-  building 'RMariaDB_1.3.1.9001.tar.gz'
   
* installing *source* package 'RMariaDB' ...
** using staged installation
Using mysql_config
Found mysql_config cflags and libs!
Using PKG_CFLAGS=-I/usr/include/mysql
Using PKG_LIBS=-L/usr/lib64/mysql -lmysqlclient -lpthread -lz -lm -ldl -lssl -lcrypto
Using PKG_PLOGR=
** libs
using C++ compiler: 'g++ (GCC) 7.3.1 20180712 (Red Hat 7.3.1-17)'
g++ -std=gnu++17 -I"/usr/lib/R/include" -DNDEBUG -I/usr/include/mysql -Ivendor -DBOOST_NO_AUTO_PTR  -I'/usr/lib/R/addons/cpp11/include' -I'/usr/lib/R/addons/plogr/include' -I/usr/local/include    -fPIC  -g -O2   -c DbConnection.cpp -o DbConnection.o
DbConnection.cpp: In member function 'void DbConnection::connect(const cpp11::sexp&, const cpp11::sexp&, const cpp11::sexp&, const cpp11::sexp&, unsigned int, const cpp11::sexp&, long unsigned int, const cpp11::sexp&, const cpp11::sexp&, const cpp11::sexp&, const cpp11::sexp&, const cpp11::sexp&, const cpp11::sexp&, const cpp11::sexp&, int, bool)':
DbConnection.cpp:47:33: error: 'MYSQL_OPT_SSL_KEY' was not declared in this scope
     mysql_options(this->pConn_, MYSQL_OPT_SSL_KEY,    cpp11::as_cpp<std::string>(ssl_key).c_str());
                                 ^~~~~~~~~~~~~~~~~
DbConnection.cpp:47:33: note: suggested alternative: 'MYSQL_OPT_USE_RESULT'
     mysql_options(this->pConn_, MYSQL_OPT_SSL_KEY,    cpp11::as_cpp<std::string>(ssl_key).c_str());
                                 ^~~~~~~~~~~~~~~~~
                                 MYSQL_OPT_USE_RESULT
DbConnection.cpp:50:33: error: 'MYSQL_OPT_SSL_CERT' was not declared in this scope
     mysql_options(this->pConn_, MYSQL_OPT_SSL_CERT,   cpp11::as_cpp<std::string>(ssl_cert).c_str());
                                 ^~~~~~~~~~~~~~~~~~
DbConnection.cpp:50:33: note: suggested alternative: 'MYSQL_OPT_USE_RESULT'
     mysql_options(this->pConn_, MYSQL_OPT_SSL_CERT,   cpp11::as_cpp<std::string>(ssl_cert).c_str());
                                 ^~~~~~~~~~~~~~~~~~
                                 MYSQL_OPT_USE_RESULT
DbConnection.cpp:53:33: error: 'MYSQL_OPT_SSL_CA' was not declared in this scope
     mysql_options(this->pConn_, MYSQL_OPT_SSL_CA,     cpp11::as_cpp<std::string>(ssl_ca).c_str());
                                 ^~~~~~~~~~~~~~~~
DbConnection.cpp:53:33: note: suggested alternative: 'MYSQL_OPT_NONBLOCK'
     mysql_options(this->pConn_, MYSQL_OPT_SSL_CA,     cpp11::as_cpp<std::string>(ssl_ca).c_str());
                                 ^~~~~~~~~~~~~~~~
                                 MYSQL_OPT_NONBLOCK
DbConnection.cpp:56:33: error: 'MYSQL_OPT_SSL_CAPATH' was not declared in this scope
     mysql_options(this->pConn_, MYSQL_OPT_SSL_CAPATH, cpp11::as_cpp<std::string>(ssl_capath).c_str());
                                 ^~~~~~~~~~~~~~~~~~~~
DbConnection.cpp:56:33: note: suggested alternative: 'MYSQL_OPT_USE_RESULT'
     mysql_options(this->pConn_, MYSQL_OPT_SSL_CAPATH, cpp11::as_cpp<std::string>(ssl_capath).c_str());
                                 ^~~~~~~~~~~~~~~~~~~~
                                 MYSQL_OPT_USE_RESULT
DbConnection.cpp:59:33: error: 'MYSQL_OPT_SSL_CIPHER' was not declared in this scope
     mysql_options(this->pConn_, MYSQL_OPT_SSL_CIPHER, cpp11::as_cpp<std::string>(ssl_cipher).c_str());
                                 ^~~~~~~~~~~~~~~~~~~~
DbConnection.cpp:59:33: note: suggested alternative: 'MYSQL_OPT_USE_RESULT'
     mysql_options(this->pConn_, MYSQL_OPT_SSL_CIPHER, cpp11::as_cpp<std::string>(ssl_cipher).c_str());
                                 ^~~~~~~~~~~~~~~~~~~~
                                 MYSQL_OPT_USE_RESULT
make: *** [DbConnection.o] Error 1
ERROR: compilation failed for package 'RMariaDB'
* removing '/home/fatkat/R/RMariaDB'
* restoring previous '/home/fatkat/R/RMariaDB'
Error: Failed to install 'RMariaDB' from GitHub:
  (converted from warning) installation of package '/tmp/RtmpIbXsV6/file72e02978f29c/RMariaDB_1.3.1.9001.tar.gz' had non-zero exit status

Enter a frame number, or 0 to exit   

1: remotes::install_github(repo = "r-dbi/RMariaDB", ref = "v1.3.1.9001", lib = "~/R", upgrade = "never", force = TRUE)
2: install_remotes(remotes, auth_token = auth_token, host = host, dependencies = dependencies, upgrade = upgrade, force = force, quiet = quiet, bu
3: tryCatch(res[[i]] <- install_remote(remotes[[i]], ...), error = function(e) {
    stop(remote_install_error(remotes[[i]], e))
})
4: tryCatchList(expr, classes, parentenv, handlers)
5: tryCatchOne(expr, names, parentenv, handlers[[1]])
6: value[[3]](cond)

Selection: 

> On Jun 30, 2024, at 4:32 AM, Kirill Müller ***@***.***> wrote:
> 
> 
> Thank you for your patience. If this is still an issue, could you test v1.3.1.9004, then v1.3.1.9002 or v1.3.1.9006, ..., to identify where the regression occurred?
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/r-dbi/RMariaDB/issues/336#issuecomment-2198481303>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ACUGAODEGC5HPAL3ZY3PDCTZJ67ALAVCNFSM6AAAAABIZCMKUOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCOJYGQ4DCMZQGM>.
> You are receiving this because you authored the thread.
>

--- a/src/DbConnection.cpp
+++ b/src/DbConnection.cpp
@@ -49,9 +49,18 @@ void DbConnection::connect(const cpp11::sexp& host, const cpp11::sexp& user,
     // MySQL 8.0 deprecated the enforce SSL flag in prefernce to defining an SSL MODE
     unsigned int ssl_mode_ = SSL_MODE_REQUIRED;
     mysql_options(this->pConn_, MYSQL_OPT_SSL_MODE, &ssl_mode_);
-#else    
+#elif MYSQL_VERSION_ID >= 50700
     my_bool use_ssl_ = 1;
     mysql_options(this->pConn_, MYSQL_OPT_SSL_ENFORCE, &use_ssl_);
+#else
+    mysql_ssl_set(
+      this->pConn_,
+      Rf_isNull(ssl_key) ? NULL : cpp11::as_cpp<std::string>(ssl_key).c_str(),
+      Rf_isNull(ssl_cert) ? NULL : cpp11::as_cpp<std::string>(ssl_cert).c_str(),
+      Rf_isNull(ssl_ca) ? NULL : cpp11::as_cpp<std::string>(ssl_ca).c_str(),
+      Rf_isNull(ssl_capath) ? NULL : cpp11::as_cpp<std::string>(ssl_capath).c_str(),
+      Rf_isNull(ssl_cipher) ? NULL : cpp11::as_cpp<std::string>(ssl_cipher).c_str()
+    );
 #endif
     // From MySQL manual: Do not set this option within an application program; it is set internally in the client library.
     // Instead the application program should set mysql_options
@@ -70,6 +79,7 @@ void DbConnection::connect(const cpp11::sexp& host, const cpp11::sexp& user,
 #endif
     client_flag &= ~CLIENT_SSL_VERIFY_SERVER_CERT;
   }
+#if MYSQL_VERSION_ID >= 50700
   if (!Rf_isNull(ssl_key)) {
     mysql_options(this->pConn_, MYSQL_OPT_SSL_KEY,    cpp11::as_cpp<std::string>(ssl_key).c_str());
   }
@@ -85,6 +95,7 @@ void DbConnection::connect(const cpp11::sexp& host, const cpp11::sexp& user,
   if (!Rf_isNull(ssl_cipher)) {
     mysql_options(this->pConn_, MYSQL_OPT_SSL_CIPHER, cpp11::as_cpp<std::string>(ssl_cipher).c_str());
   }
+#endif
   if (timeout > 0) {
     mysql_options(this->pConn_, MYSQL_OPT_CONNECT_TIMEOUT,
                   &timeout);

